src/sql/parts/registeredServer/common/objectExplorerService.ts

```diff
@@ -65,16 +65,29 @@ export interface IObjectExplorerService {
 	onSelectionOrFocusChange: Event<void>;
 
 	getServerTreeView(): ServerTreeView;
+
+	getActiveConnections(): { connectionId: string, nodeInfo: sqlops.NodeInfo }[];
+
+	getChildren(connectionId: string, nodePath: string): Thenable<sqlops.NodeInfo[]>;
```

Comment:
Have we looked at the built-in [TreeDataProvider APIs](https://code.visualstudio.com/docs/extensionAPI/vscode-api#TreeDataProvider) and considered how we can leverage their patterns / improve on their design? 
- For contributors to the tree, they have a pattern of "getChildren returns a set of objects, and given the objects you can request a TreeItem using getTreeItem".
- In this case it seems NodeInfo is a lot like TreeItem, but it can't be abstracted away depending on the node provider.

Similarly they have a way to provide tri-state expansion (expanded/collapsed/node, where node means it's a leaf node). 

What VSCode seems to lack is programmatic support for "select this node" etc. so it's great we're adding this in. The closer we can align the easier for contributors to use our code & for us to potentially contribute back to VSCode down the line.

Reply:
I refactored things a bit to use the VS Code expansion enum and make the rest of the code match up a bit more with their way of doing things. In the future we may want to refactor things a bit more if we end up exposing similar APIs for other trees, but that should be fairly straightforward now